db_stress_tool/db_stress_env_wrapper.h

```diff
@@ -13,6 +13,23 @@
 #include "monitoring/thread_status_util.h"
 
 namespace ROCKSDB_NAMESPACE {
+namespace {
+void CheckIOActivity(const IOOptions& options) {
+#ifndef NDEBUG
+  const ThreadStatus::OperationType thread_op =
+      ThreadStatusUtil::GetThreadOperation();
+  Env::IOActivity io_activity =
+      ThreadStatusUtil::TEST_GetExpectedIOActivity(thread_op);
+  // TODO: fix places where default IOOption() is used, which can cause
+  // io_activity to not match thread operation.
+  assert(io_activity == Env::IOActivity::kUnknown ||
+         options.io_activity == Env::IOActivity::kUnknown ||
```

Comment:
We kind of rely on NOT having this exception "options.io_activity == Env::IOActivity::kUnknown" to find incorrectly passed in IOOptions.... Is there anyway to fix the UT instead? By the way I like the refactory.

Reply:
I could change the sync point used in unit test to make it work. But I think it makes sense to keep the thread_operation as Flush/Compaction instead of setting it to UNKNOWN. I assume that setting the thread_operation to UNKNOWN is only there to pass the checks in stress test. One alternative is to set the IOActivity in ~WritableFileWriter() based on thread_operation, what do you think?

Reply:
> But I think it makes sense to keep the thread_operation as Flush/Compaction instead of setting it to UNKNOWN

Right unfortunately we can't pass parameter into destructor

> I assume that setting the thread_operation to UNKNOWN is only there to pass the checks in stress test. 

Right 

> One alternative is to set the IOActivity in ~WritableFileWriter() based on thread_operation, what do you think?

Yeah without refactoring the cleanup out of destructor you may do that under dbg mode.

Reply:
Updated to set IOActivity in destructor.